src/serializer/ResidualHeapSerializer.js

```diff
@@ -695,6 +701,120 @@ export class ResidualHeapSerializer {
     return t.arrayExpression(initProperties);
   }
 
+  _getPropertyValue(properties: Map<string, any>, key: string) {
+    if (properties.has(key) === true) {
+      let val = properties.get(key);
+
+      if (val !== undefined) {
+        let descriptor = val.descriptor;
+
+        if (descriptor !== undefined) {
+          let descriptorValue = descriptor.value;
+
+          if (descriptorValue !== undefined) {
+            return descriptorValue;
+          }
+        }
+      }
+    }
+    return null;
+  }
+
+  _serializeValueReactElementChild(child: Value): BabelNode {
+    if (child instanceof ObjectValue && child.properties.has("$$typeof")) {
+      // if we know it's a ReactElement, we add the value to the serializedValues
+      // and short cut to get back the JSX expression so we don't emit additional data
+      // we do this to ensure child JSXElements can get keys assigned if needed
+      this.serializedValues.add(child);
+      return this._serializeValueObject(child);
+    }
+    const expr = this.serializeValue(child);
+
+    if (t.isArrayExpression(expr)) {
+      applyKeysToNestedArray(((expr: any): BabelNodeArrayExpression), true, this.react.usedReactElementKeys);
+    } else if (t.isStringLiteral(expr) || t.isNumericLiteral(expr)) {
+      return t.jSXText((expr: any).value + "");
```

Comment:
The :any cast would hide any spelling error in `value`. Cast to the actual types.